### DISCUSSION TITLE
How to trigger a render from inside the script sent to the web worker.

### DISCUSSION THREAD

**Discussion by mmiscool** [zalo/CascadeStudio#85] [Q&A]
I have a timer that I want to run after a 30 seconds that adds a new body. 
The console.log() is triggered with the setTimeout function but I won't see the second shape unless I hit evaluate. 

```
  var box;
  var newbox;

async function dothisAfterLoaingUpThecode(){
   box    =  Translate([0, 0, 20],  Box(100, 50, 10, true));
  if (Checkbox("Loft?", true)){ box =   FilletEdges(box,3,[6]) };
  setTimeout(func, 30 * 1000);
}


const func = () => {
  console.log("doing the fillet");
  newbox = Translate([0, 0, 20],FilletEdges(box,4,[8]));
//Need to trigger a re render after the new fillet is made.
};
dothisAfterLoaingUpThecode();

console.log("hello world");
```

**Comment by zalo** [comment]
Unfortunately, this whole pipeline wasn't written with async in mind.

If you're trying to retrieve a library over the internet, I've hacked it in a [PR branch as a blocking XMLHttpRequest](https://github.com/zalo/CascadeStudio/pull/15/files#diff-7d888ae3a42f72ffc3319a8a7a9476c873382cdfa729f59ef60b2c5ea257d44cR35-R55).

If you're feeling evil, calling this _may_ trigger a re-render...
`messageHandlers["combineAndRenderShapes"]({ "maxDeviation": 0.01 });`
